lnwallet/channel.go

```diff
@@ -4573,6 +4720,60 @@ func (lc *LightningChannel) ReceiveRevocation(revMsg *lnwire.RevokeAndAck) (
 		}
 	}
 
+	var localPeerUpdates []channeldb.LogUpdate
+
+	// We use the remote commitment chain's tip as it will soon become the tail
+	// once advanceTail is called.
+	remoteMessageIndex := lc.remoteCommitChain.tip().ourMessageIndex
+	localMessageIndex := lc.localCommitChain.tail().ourMessageIndex
+
+	for e := lc.localUpdateLog.Front(); e != nil; e = e.Next() {
+		pd := e.Value.(*PaymentDescriptor)
+
+		if pd.LogIndex < remoteMessageIndex && pd.LogIndex >= localMessageIndex {
+			// This is a settle/fail that is on the remote commitment, but
+			// not on the local commitment. We expect this update to be
+			// covered in the next commitment signature the remote sends.
+			logUpdate := channeldb.LogUpdate{
+				LogIndex: pd.LogIndex,
+			}
+
+			// don't save adds
+			if pd.EntryType == Add {
```

Comment:
This can be moved to reside outside the if statement so we can skip them early.

Reply:
done